app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -64,14 +144,65 @@ class PipelineViz extends React.Component {
     return stagesWithModifiedNames;
   }
 
+  handleMouseWheelZoom = e => {
+    const { zoomChangeInterval } = this.props;
+    const zoomChange = (e.deltaY < 0 ? 1 : -1) * zoomChangeInterval;
+    this.setState({ zoom: this.state.zoom + zoomChange });
+  };
+
+  handleStepClick(stageIndex, info) {
+    const clickedNodeId = info.nodes[0];
+    if (clickedNodeId == null) {
```

Comment:
could also do `if (!clickedNodeId) return;`

Reply:
I tried that but there are situations where `clickedNodeId` is 0 : ' (

Reply:
nit: this actually comparing clickedNodeId to null and undefined... which should be fine.

Reply:
Visjs actually calls this each time the graph is clicked (not just the node), so there may be situations `clickedNodeId` is undefined (hence the check)